### PR TITLE
Return error if data file can't be deleted

### PR DESCRIPTION
**Pull request by CalebFenton**
Also, standardize rootdirs and add missing datadao config
to local.conf.template.

### CODE REVIEW

**Comment by CalebFenton** [comment]
I wasn't sure how to best add tests for this. Advice welcome.

File: job-server/src/spark.jobserver/io/DataFileDAO.scala [hunk 1]
```diff
@@ -95,13 +95,14 @@ class DataFileDAO(config: Config) {
     out.writeLong(aInfo.uploadTime.getMillis)
   }
 
-  def deleteFile(aName: String) {
+  def deleteFile(aName: String): Boolean = {
     if (aName.startsWith(rootDir) && files.contains(aName)) {
       // only delete the file if it is known to this class,
       // otherwise this could be abused
-      new File(aName).delete
       files -= aName
```
Comment (noorul) [inline job-server/src/spark.jobserver/io/DataFileDAO.scala]: I think this should happen after successfully deleting the file. What if the below line throws an exception?

**Comment by noorul** [comment]
@TimMaltGermany Was it a security decision to always return OK ?

File: job-server/src/spark.jobserver/io/DataFileDAO.scala (hunk 1 above)
Comment (CalebFenton) [inline job-server/src/spark.jobserver/io/DataFileDAO.scala]: That's fine.

File: job-server/src/spark.jobserver/DataManagerActor.scala [hunk 2]
```diff
@@ -25,13 +25,11 @@ class DataManagerActor(fileDao: DataFileDAO) extends InstrumentedActor {
   override def wrappedReceive: Receive = {
     case ListData => sender ! fileDao.listFiles
 
-    case DeleteData(fileName) => {
-      fileDao.deleteFile(fileName)
-      sender ! Deleted
-    }
+    case DeleteData(fileName) =>
+      if (fileDao.deleteFile(fileName)) sender ! Deleted else sender ! Error
```
Comment (velvia) [inline job-server/src/spark.jobserver/DataManagerActor.scala]: `sender ! { if (fileDao.deleteFile(fileName)) Deleted else Error }

**Comment by velvia** [comment]
@CalebFenton thanks.  I'm fine with this change, other than a minor comment.

**Comment by velvia** [comment]
@CalebFenton thanks.  Going to merge this.

**Pull request merged**